src/components/Header/index.tsx

```diff
@@ -1,7 +1,20 @@
 import { Box, Flex, Heading, Input, Text } from "@chakra-ui/react";
+import { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useQuery } from "../../hooks/useQuery";
 import { Logo } from "../../icons/Logo";
 
 export function Header() {
+  const q = useQuery().get("q") ?? "";
+  const [searchValue, setSearchValue] = useState(q);
+  const history = useHistory();
+  const onSearchChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
+    setSearchValue(evt.target.value);
+  };
+  const onSearchSubmit = (evt: React.FormEvent): void => {
+    evt.preventDefault();
+    history.push(`/search?q=${searchValue}`);
+  };
```

Comment:
We may eventually want to break out the search logic & ui into a separate `<Search />` component

Reply:
yeah I agree, lets do it with the UI implementation?